example/offboard_velocity/offboard_velocity.cpp

```diff
@@ -211,6 +230,31 @@ void usage(std::string bin_name)
               << "For example, to connect to the simulator use URL: udp://:14540" << std::endl;
 }
 
+Telemetry::landed_state_callback_t landed_state_callback(std::shared_ptr<Telemetry>& telemetry, std::promise<void>& landed_promise)
+{
+    return [&landed_promise, &telemetry](Telemetry::LandedState landed) {
+        switch (landed) {
+            case Telemetry::LandedState::ON_GROUND:
+                std::cout << "On ground" << std::endl;
+                break;
+            case Telemetry::LandedState::TAKING_OFF:
+                std::cout << "Taking off..." << std::endl;
+                break;
+            case Telemetry::LandedState::LANDING:
+                std::cout << "Landing..." << std::endl;
+                break;
+            case Telemetry::LandedState::IN_AIR:
+                std::cout << "Taking off has finished." << std::endl;
+                landed_promise.set_value_at_thread_exit();
```

Comment:
What is this? I've never seen it.

Why don't you just do:
```
telemetry->landed_state_async(nullptr);
landed_promise.set_value();
```

Reply:
Actually I had called `telemetry->landed_state_async(nullptr);` after the `set_value()` in line 249.
I will try to call async before setting the value